File: pyramide.c
#include <stdio.h>
#include <sys/ioctl.h>
#include <unistd.h>

int main(){
	struct winsize w;
	ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
	int width = w.ws_col/2 - 1;
	int um = 0, n = 0;
	int pyramide = 1;
	int reverse = 0;
	for(int times = 0; times <= n; times++){
		if (pyramide){
			for(int i = um; i <= width; i++){
				for(int idk = 0; idk <= width - i - 1; idk++){
					printf(" ");
				}
				for(int j = 1; j <= i + 1; j++){
					printf("* ");
				}
				printf("\n");
			}
		}
		if (reverse){
			for(int i = width - 1; i >= 0; i=i-1){
		                for(int idk = 0; idk <= width - i - 1; idk++){
		                        printf(" ");
		                }
		                for(int j = 1; j <= i + 1; j++){
		                        printf("* ");
		                }
		                printf("\n");
		        }
			int um = 1;
		}
	}
	return 0;
}

File: pyramide.py
import os

width = int(list(os.get_terminal_size())[0]/2.2)

pyramide = True
reverse = False

n = 1

for times in range(n):
	if pyramide:
		for i in range(width):
			print(" ", " ")
			for idk in range(width - int(i)):
				print(" ", end='')
			for j in range(i + 1):
				print("* ", end='')
	if reverse:
		for i in range(width - 1):
			print(" ", " ")
			i = width - i
			for idk in range(width - i):
				print(' ', end='')
			for j in range(0, i):
				print("* ", end='')
		print('\n')
